### ISSUE TITLE
2021.6 Errors in Log

### ISSUE THREAD

**Issue by kreene1987** [basnijholt/adaptive-lighting#128]
Still digesting, but trying to figure out what changed:

```
Logger: homeassistant.components.switch
Source: custom_components/adaptive_lighting/switch.py:1045
Integration: Switch (documentation, issues)
First occurred: 1:53:37 PM (8 occurrences)
Last logged: 1:53:37 PM

Error adding entities for domain switch with platform adaptive_lighting
Error while setting up adaptive_lighting platform for switch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 383, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 588, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 614, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/adaptive_lighting/switch.py", line 634, in async_added_to_hass
    await self.async_turn_on(adapt_lights=not self._only_once)
  File "/config/custom_components/adaptive_lighting/switch.py", line 723, in async_turn_on
    await self._update_attrs_and_maybe_adapt_lights(
  File "/config/custom_components/adaptive_lighting/switch.py", line 851, in _update_attrs_and_maybe_adapt_lights
    self._settings = self._sun_light_settings.get_settings(
  File "/config/custom_components/adaptive_lighting/switch.py", line 1146, in get_settings
    percent = self.calc_percent()
  File "/config/custom_components/adaptive_lighting/switch.py", line 1109, in calc_percent
    today = self.relevant_events(now)
  File "/config/custom_components/adaptive_lighting/switch.py", line 1097, in relevant_events
    events = [
  File "/config/custom_components/adaptive_lighting/switch.py", line 1098, in <listcomp>
    self.get_sun_events(now + timedelta(days=days)) for days in [-1, 0, 1]
  File "/config/custom_components/adaptive_lighting/switch.py", line 1053, in get_sun_events
    else _replace_time(date, "sunrise")
  File "/config/custom_components/adaptive_lighting/switch.py", line 1045, in _replace_time
    utc_time = self.time_zone.localize(date_time).astimezone(dt_util.UTC)
AttributeError: 'str' object has no attribute 'localize'
```

```
This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/adaptive_lighting/switch.py:1045
Integration: Adaptive Lighting
First occurred: 1:57:14 PM (**253 occurrences**)
Last logged: 3:30:41 PM

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/adaptive_lighting/switch.py", line 738, in _async_update_at_interval
    await self._update_attrs_and_maybe_adapt_lights(
  File "/config/custom_components/adaptive_lighting/switch.py", line 851, in _update_attrs_and_maybe_adapt_lights
    self._settings = self._sun_light_settings.get_settings(
  File "/config/custom_components/adaptive_lighting/switch.py", line 1146, in get_settings
    percent = self.calc_percent()
  File "/config/custom_components/adaptive_lighting/switch.py", line 1109, in calc_percent
    today = self.relevant_events(now)
  File "/config/custom_components/adaptive_lighting/switch.py", line 1097, in relevant_events
    events = [
  File "/config/custom_components/adaptive_lighting/switch.py", line 1098, in <listcomp>
    self.get_sun_events(now + timedelta(days=days)) for days in [-1, 0, 1]
  File "/config/custom_components/adaptive_lighting/switch.py", line 1053, in get_sun_events
    else _replace_time(date, "sunrise")
  File "/config/custom_components/adaptive_lighting/switch.py", line 1045, in _replace_time
    utc_time = self.time_zone.localize(date_time).astimezone(dt_util.UTC)
AttributeError: 'str' object has no attribute 'localize'
```

**Comment by kreene1987** [comment]
Looks like "time" was updated by @spacegaier from HA devs. I'll continue to pursue.

**Comment by spacegaier** [comment]
> 
> 
> Looks like "time" was updated by @spacegaier from HA devs. I'll continue to pursue.

That is not related to my changes. Mine were in the HA frontend, so nothing to do with Python files.

But there were also backend timezone changes that might come into play here.

**Comment by LeChuck2909** [comment]
I also noticed, that some of my adaptive light integrations not work anymore. I found, that the one where sunrise_time is still "None", works, the other one where sunrise_time was 07:00:00 did not work anymore. by removing the 07:00:00 and replace it with "None" again, i bring all adpative light integrations to work again. so i guess the issue is related to sunrise_time etc...

**Comment by yurnih** [comment]
I got the same issue after upgrading. See also home-assistant/core#49643, with that merge the pytz timezone object changed to a string. I figured out that with a change of line [1045](https://github.com/basnijholt/adaptive-lighting/blob/a577a1953d98acadb462041bf4d3503db0b87ed4/custom_components/adaptive_lighting/switch.py#L1045) to `utc_time = date_time.replace(tzinfo=dt_util.DEFAULT_TIME_ZONE).astimezone(dt_util.UTC)` the issue got solved for 2021.6.

**Comment by basnijholt** [comment]
Thanks @yurnih! I have used your fix in https://github.com/basnijholt/adaptive-lighting/commit/40bc5dad17045facfa2bb5f384591aeea0256e9e.

Anyone, please let me know whether this fixed the issue. If not I will reopen this issue.
